### PR TITLE
[user_login] finish #3.

### LINKED ISSUE #3: Implement sign up and log in

### PR DESCRIPTION
**Pull request by xingweitian**
resolves #3 .

About UserLogin:

For safety consider, I think we shouldn't take the whole user object (include **password**) in UserLogin object. 
So I optimize it. When creating the class instance of UserLogin, just put id, email and type into the instance.

What have I done?

- [x] Use random strings as secret key to increase security.

- [x] Function: Log in.

- [x] Function: Sign up.

### CODE REVIEW

File: utils/error_msg.py [hunk 1]
```diff
@@ -4,3 +4,4 @@
 # Custom Message
 ILLEGAL_ARGUMENT = MSG_400
 DUPLICATE_EMAIL = 'User with the same email exists'
+WRONG_PASSWORD_OR_EMAIL_ADDRESS = 'invalid email address or password'
```
Comment (zhangjiangqige) [inline utils/error_msg.py]: "Invalid"; better use the capital case to be in the same style as other errors

File: views/user.py [hunk 2]
```diff
@@ -35,3 +44,29 @@ def register():
...
+        u = UserLogin(str(user._id), user.email, user.type)
+        login_user(user=u, remember=remember_me)
+        return jsonify({
+            "success": 1
+        })
+    except Exception as e:
+        logger.exception("Exception happened when trying to login. {}".format(e))
+        return jsonify({
+            "success": 0
+        }), 500
```
Comment (zhangjiangqige) [inline views/user.py]: let's use the global 500 handler in dingziku

File: views/user.py [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
-
-import logging
-
+from utils.login_utils import UserLogin
 
 logger = logging.getLogger(__name__)
 user_blueprint = Blueprint("user", __name__)
 
 
+def str_to_boolean(string):
```
Comment (zhangjiangqige) [inline views/user.py]: move to utils

**Review by zhangjiangqige — CHANGES_REQUESTED** [review]
see comments

**Comment by xingweitian** [comment]
@neodooth, please check again.

File: new_airbnb.py [hunk 4]
```diff
@@ -4,10 +4,16 @@
 from flask import Flask
 from flask_cors import CORS
 
+from utils.error_handler import generate
+
 app = Flask(__name__)
 app.secret_key = "".join(random.choice(string.ascii_letters + string.digits) for _ in range(15))
 CORS(app)
 
+error_codes = [400, 401, 403, 404, 405, 409, 413, 500]
```
Comment (zhangjiangqige) [inline new_airbnb.py]: I think these codes would be better to be placed in error_msg.py so it's easier to check if every status code has a corresponding message.

Also registering these error handlers is not the job of this python file; its only purpose is to store `app` for other modules to import and we should keep this file away from our business logic as far as possible.

I suggest we call `generate` in error_handler.py (as an initialization), and import error_handler in \_\_init\_\_ of views; error codes are more related to views than to the app itself logically.

**Review by zhangjiangqige — COMMENTED** [review]
about registering error handlers

**Review by zhangjiangqige — APPROVED** [review]

**Pull request merged**